sdk/dapp-kit/src/walletStore.ts

```diff
@@ -6,8 +6,14 @@ import { createStore } from 'zustand';
 import type { StateStorage } from 'zustand/middleware';
 import { createJSONStorage, persist } from 'zustand/middleware';
 
+type WalletConnectionStatus = 'disconnected' | 'connecting' | 'connected';
+
+type WalletAutoConnectionStatus = 'disabled' | 'idle' | 'settled';
```

Comment:
I just slapped this together for reference sake before I sign off and you head out on PTO @Jordan-Mysten. Is this what you were thinking, or should we have more granular "error", "loading", "success" statuses here instead of the super basic three I included?

this gets a little weird if the top-level wallet provider component re-renders and then the auto connect mutation runs again, but I think for this case we really only care about the first run?

I could also do something like https://github.com/TanStack/query/issues/2304#issuecomment-1203942405 which more closely mimics the `useMutationState` hook in v5

Reply:
I think this is totally fine, the sub-states shouldn't matter and can be derived.